### ISSUE TITLE
`Dict` object returns different exceptions for the missing keys.

### ISSUE THREAD

**Issue by yakutovicha** [aiidateam/aiida-core#4454]
### Describe the bug
For the dictionary object created like so `d = Dict(dict={'a':{'b':1}})` different exceptions are raised for `d['x']` and for `d['a']['x']`:
- `d['x']` raises `AttributeError`
-  `d['a']['x']` raises `KeyError`

The final result is that 
```python
try:
    d['x']
except KeyError:
    print("Test")
```
Can't catch the error. 

### Steps to reproduce

From the `verdi shell`:
```python
In [1]: from aiida.orm import Dict

In [2]: d = Dict(dict={'a':{'b':1}})

In [3]: d['x']
---------------------------------------------------------------------------
KeyError                                  Traceback (most recent call last)
/opt/conda/lib/python3.7/site-packages/aiida/orm/implementation/entities.py in get_attribute(self, key)
    153         try:
--> 154             return self._dbmodel.attributes[key]
    155         except KeyError as exception:

KeyError: 'x'

The above exception was the direct cause of the following exception:

AttributeError                            Traceback (most recent call last)
<ipython-input-3-7657742692bd> in <module>
----> 1 d['x']

/opt/conda/lib/python3.7/site-packages/aiida/orm/nodes/data/dict.py in __getitem__(self, key)
     63
     64     def __getitem__(self, key):
---> 65         return self.get_attribute(key)
     66
     67     def __setitem__(self, key, value):

/opt/conda/lib/python3.7/site-packages/aiida/orm/entities.py in get_attribute(self, key, default)
    343         """
    344         try:
--> 345             attribute = self.backend_entity.get_attribute(key)
    346         except AttributeError:
    347             if default is _NO_DEFAULT:

/opt/conda/lib/python3.7/site-packages/aiida/orm/implementation/entities.py in get_attribute(self, key)
    154             return self._dbmodel.attributes[key]
    155         except KeyError as exception:
--> 156             raise AttributeError('attribute `{}` does not exist'.format(exception)) from exception
    157
    158     def get_attribute_many(self, keys):

AttributeError: attribute `'x'` does not exist
```

### Expected behavior

`d['x']` should except with the `KeyError`
### Your environment

- Operating system Linux:
- Python 3.7.4:
- aiida-core 1.4.2:

**Comment by greschd** [comment]
Yep, the `__getitem__` and `__setitem__` should try / except and re-raise with the correct exception.

Introduced in https://github.com/aiidateam/aiida-core/pull/4351, tagging @sphuber @ramirezfranciscof

**Comment by sphuber** [comment]
Fixed in PR #4616

**Comment by greschd** [comment]
@sphuber isn't this still open in the current `develop` (and fixed only in the `1.4.x` release branch)?

**Comment by sphuber** [comment]
We are in the process of releasing `1.4.4` and `1.5.1` now and `1.5.1` will be merged into develop. Since I assign these fixes to a milestone, corresponding to the version in which they will be released, I close them as soon as they are merged in that release/support branch.
